crates/ethcore/types/src/views/typed_transaction.rs

```diff
@@ -0,0 +1,246 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+// This file is part of OpenEthereum.
+
+// OpenEthereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// OpenEthereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with OpenEthereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! View onto transaction rlp
+use std::convert::TryInto;
+
+use crate::transaction::{signature, TypedTxId};
+
+use super::ViewRlp;
+use bytes::Bytes;
+use ethereum_types::{H256, U256};
+use hash::keccak;
+
+/// View onto transaction rlp. Assumption is this is part of block.
+/// Typed Transaction View. It handles raw bytes to search for particular field.
+/// Access tx:
+/// 1 | [chainId, nonce, gasPrice, gasLimit, to, value, data, access_list, senderV, senderR, senderS]
+/// Legacy tx:
+/// [nonce, gasPrice, gasLimit, to, value, data, senderV, senderR, senderS]
+pub struct TypedTransactionView<'a> {
+    rlp: ViewRlp<'a>,
+}
```

Comment:
```suggestion
pub struct TypedTransactionView<'a> {
    rlp: ViewRlp<'a>,
    type: TypedTxId,
}
```

If we want to call the mehods `nonce`, `gas_price`, etc... the `transaction_type()` will be called multiple times with some computation.

Why not to get the transaction type and set into the `TypedTransactionView::new(...)` ?

Reply:
It makes sense, will change it.